### PR TITLE
chore(cli): env init error when using vpc with no dns

### LINKED ISSUE #1332: Should error if imported VPC doesn't enable DNS support
Right now if we create a Copilot environment with disabled DNS support, then we cannot deploy services in that environment because we won't be able to resolve that DNS and have no access to the Amazon ECR endpoint.

For now we should probably just error out when we find out that the VPC they are trying to import doesn't have DNS support and give them a nice reason for that.

### PR DESCRIPTION
**Pull request by iamhopaul123**
fixes #1332

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/aws/ec2/ec2.go [hunk 1]
```diff
@@ -153,6 +154,18 @@ func (c *EC2) ListVPCs() ([]VPC, error) {
 	return vpcs, nil
 }
 
+// VPCWithDNSSupport returns if DNS resolution is enabled for the VPC.
+func (c *EC2) VPCWithDNSSupport(vpcID string) (bool, error) {
```
Comment (efekarakus) [inline internal/pkg/aws/ec2/ec2.go]:
```suggestion
func (c *EC2) HasDNSSupport(vpcID string) (bool, error) {
```

File: internal/pkg/aws/ec2/ec2.go [hunk 2]
```diff
@@ -153,6 +154,18 @@ func (c *EC2) ListVPCs() ([]VPC, error) {
...
+	resp, err := c.client.DescribeVpcAttribute(&ec2.DescribeVpcAttributeInput{
+		VpcId:     aws.String(vpcID),
+		Attribute: aws.String(ec2.VpcAttributeNameEnableDnsSupport),
+	})
+	if err != nil {
+		return false, fmt.Errorf("describe attribute for VPC %s: %w", vpcID, err)
```
Comment (efekarakus) [inline internal/pkg/aws/ec2/ec2.go]: nit:
```suggestion
		return false, fmt.Errorf("describe %s attribute for VPC %s: %w", ec2.VpcAttributeNameEnableDnsSupport, vpcID, err)
```

File: internal/pkg/cli/env_init.go [hunk 3]
```diff
@@ -365,6 +366,20 @@ func (o *initEnvOpts) askImportResources() error {
...
+	}
+	dnsSupport, err := o.ec2Client.VPCWithDNSSupport(o.ImportVPC.ID)
+	if err != nil {
+		return fmt.Errorf("check if VPC %s has DNS support enabled: %w", o.ImportVPC.ID, err)
+	}
+	if !dnsSupport {
+		log.Warningln(`Looks like you're creating an environment using a VPC with DNS support disabled.
+However, currently Copilot will not be able to deploy a service then.
+We recommend using VPC with enabled DNS. To learn more about the issue:
+https://aws.amazon.com/premiumsupport/knowledge-center/ecs-pull-container-api-error-ecr/`)
```
Comment (efekarakus) [inline internal/pkg/cli/env_init.go]:
```suggestion
		log.Errorln(`Looks like you're creating an environment using a VPC with DNS support *disabled*.
Copilot cannot create services in VPCs without DNS support. We recommend enabling this property.
To learn more about the issue:
https://aws.amazon.com/premiumsupport/knowledge-center/ecs-pull-container-api-error-ecr/`)
```

**Review by efekarakus — COMMENTED** [review]
Looks good!

**Review by efekarakus — APPROVED** [review]

**Review by kohidave — APPROVED** [review]

**Pull request merged**